tests/unit/test_log_sanitizer.py

```
@@ -0,0 +1,244 @@
+        assert sanitize_data(None) is None
+
+    def test_empty_dict(self):
+        assert sanitize_data({}) == {}
+
+    def test_empty_list(self):
+        assert sanitize_data([]) == []
+
+    def test_does_not_mutate_original(self):
+        data = {"password": "secret", "name": "test"}
+        result = sanitize_data(data)
+        assert data["password"] == "secret"
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to: "password"

(S105)